File: LCD1602_I2C_Module_Demo/Jetson Nano/python/examples/Direction_control.py
import sys
sys.path.append('../lib/')
import LCD1602  # Import the LCD1602 module
import time  # Import the time module for delays

# Initialize LCD with 16 columns and 2 rows
lcd = LCD1602.LCD1602(16, 2)
lcd.cursor()  # Enable cursor on the LCD

# Initialize the backlight using SN3193
led = LCD1602.SN3193()

# Set the backlight brightness to 50% (range: 0~100)
led.set_brightness(50)

# Text to be displayed
text = "Waveshare"
text1 = "Hello World!"

try:
    # Set the cursor to column 0, row 0
    lcd.setCursor(0, 0)
    lcd.leftToRight()  # Set text direction from left to right

    # Display the text "Waveshare" on the first row
    for i in range(9):  # Loop through the first text
        lcd.data(ord(text[i]))  # Send character to LCD
        time.sleep(0.2)  # Wait for 0.1 second

    lcd.rightToLeft()  # Change text direction to right to left

    # Set initial position for scrolling text
    x = 39  # The X-coordinate of the last pixel
    for i in range(11,-1,-1):  # Loop through the second text
        if i < 2:
            # Scroll display to the right
            lcd.scrollDisplayRight()
            lcd.setCursor(x, 0)  # Set cursor to the updated X position
            x = x - 1  # Move cursor left for each new character
        
        # Display characters of the second text
        lcd.data(ord(text1[i]))
        time.sleep(0.2)  # Wait for 0.1 second
        
    # Uncomment the following lines to enable breathing mode for the backlight
    # led.set_mode(LCD1602.LED_BREATH_MODE)
    # time.sleep(0.1)

except KeyboardInterrupt:  # Handle user interruption (Ctrl+C)
    # Clear the LCD display
    lcd.clear()
    # Delete the LCD object to free resources
    del lcd

File: LCD1602_I2C_Module_Demo/Pico/c/examples/Direction_control.c
/*****************************************************************************
* | File      	:   Direction_control.c
* | Author      :   Waveshare team
* | Function    :   
* | Info        :
*----------------
* | This version:   V1.0
* | Date        :   2025-01-03
* | Info        :
#
# Permission is hereby granted, free of charge, to any person obtaining a copy
# of this software and associated documentation files (the "Software"), to deal
# in the Software without restriction, including without limitation the rights
# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
# copies of the Software, and to permit persons to whom the Software is
# furnished to do so, subject to the following conditions:
#
# The above copyright notice and this permission notice shall be included in
# all copies or substantial portions of the Software.
#
# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
# LIABILITY WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
# THE SOFTWARE.
#
******************************************************************************/ 

#include "LCD_Test.h"

// Function to display basic text on the LCD screen
void Direction_control(void)
{
    // Initialize the DEV module, return if initialization fails
    if(DEV_Module_Init() != 0) {
        return;
    }

    // Output message to indicate demo
    printf("Direction control Demo\r\n");
    
    /* Initialize LCD screen with 16 columns and 2 rows */
    LCD1602_init(16, 2);
    
    // Set the backlight brightness to 50% (range: 0 to 100)
    LCD1602_set_brightness(50);

    LCD1602_cursor();  // Enable cursor on the LCD

    // Define the text to be displayed
    char text[10] = "Waveshare";  // First text to display
    char text1[13] = "Hello World!";  // Second text to scroll
    
    LCD1602_setCursor(0, 0);
    LCD1602_leftToRight();  // Set text direction to left-to-right

    // Display the text "Waveshare" on the first row
    for (int i = 0; i < 9; i++) {  // Loop through the "Waveshare" text
        LCD1602_data(text[i]);  // Send character to LCD
        DEV_Delay_ms(200);  // Wait for 0.2 seconds before displaying next character
    }

    LCD1602_rightToLeft();  // Change text direction to right-to-left for scrolling

    // Set initial position for scrolling text
    uint8_t x = 39;  // The X-coordinate for the last pixel (out of screen)
    for (int i = 11; i >= 0; i--) {  // Loop through the "Hello World!" text (backwards)
        if (i < 2) {
            // Scroll display to the right when near the left edge
            LCD1602_scrollDisplayRight();
            LCD1602_setCursor(x, 0);  // Set cursor to the updated X position
            x = x - 1;  // Move cursor left for each new character
        }
        // Display the character of the second text
        LCD1602_data(text1[i]);
        DEV_Delay_ms(200);  // Wait for 0.2 seconds before displaying next character
    }

    // Optional: Uncomment the next line to set the LED backlight to breathing mode
    // LCD1602_set_led_mode(LED_BREATH_MODE);
}
